notes/NewSiteArchitecture.md

```diff
@@ -0,0 +1,174 @@
+
+* Fragments describe a portion of some Type to be queried.
+* Fragments are often used next to View code where data is used.
+* Fragments are composed together to create full queries.
+
+5) GraphQL makes backwards-compatible APIs easy.
+
+* Server does not need to worry about concerns of any particular client, only the complete set of capabilities. Clients are responsible for the data they receive.
+* Because GraphQL only sends what you ask for, new capabilities can be introduced via new fields on types with no impact on existing queries.
+* Old capabilities can be marked "deprecated" with no impact on existing queries.
+* Far fewer reasons for versioning your API when using GraphQL.
+
+6) GraphQL queries are answered by simple functions on your server.
+
+* GraphQL is not backed by any database technology, just like REST.
+* Every field on each type is represented by a function for retrieving that data.
+* A GraphQL executor will call your functions and use as much concurrency as possible.
+* Possibilities are endless for connecting to your server's existing data model.
+
+Finally, there will be a set of links for learning more (diving into `Learn`) and for getting started (in `Code`). As well as a wall-o-logo for companies using GraphQL.
+
+## Learn
+
+*Goal:* Introduce GraphQL, one concept at a time, covering both primary concepts and best practices.
+
+*Timeframe:* Basic primary concepts by Sept 12th, advanced primary concepts by Sept 30th, best practices as ready over Q3/Q4.
+
+Where "GraphQL the Spec" is designed for a specific audience of those building GraphQL servers, this represents "GraphQL the Book" and is designed for the audience of anyone who wishes to use GraphQL. It should cover both GraphQL core concepts in addition to best practices and further topics, and it should range from introductory concepts through advanced concepts.
+
+The landing page for this section should begin as a more information-rich introduction to GraphQL, explaining why you might use it, and give a brief overview of the constituent parts. Take-aways from this introduction page:
+
+* GraphQL is a query language.
+* GraphQL servers describe a type system, called a "schema".
+* Clients can access a GraphQL server's type system to learn about what's possible.
+* Clients send queries to servers and typically get back JSON.
+* GraphQL servers validate and execute GraphQL queries.
+
+There is then a TOC through the sections and chapters (this is a straw-man list, open to reordering and addition)
+
+* Core Concepts:
+  * Query Language:
+    * Basics (queries, fields, arguments, aliases, comments)
+    * Variables
+    * Fragments
+    * Values
+    * Directives (skip & include)
+  * Type System:
+    * Basics (Schema, Objects & Fields)
+    * Scalars
+    * Lists & NonNull
+    * Interfaces
+    * Unions
+    * Enums
+  * Advanced Topics:
+    * Validation
+    * Execution & Error Handling
+    * Introspection
+    * Mutations
```

Comment:
Agreed. Mutations tends to get de-emphasized because there's so little of the GraphQL surface area that needed to be built specifically for mutations.

I think mutations should be discussed throughout these docs whenever we're talking about queries but specifically introduced in two parts:

First, somewhere in the first couple chapters. Essentially as soon as we introduce `query Foo {}` we should immediately follow with explaining `mutation Foo {}` - that should give a summation of the differences between the two.

Second, is when we explain query execution, we should explain mutations there as well.

Perhaps this is more evidence that these first three sections should be interleaved. Execution, Error Handling, and Mutations are not really "advanced" per se.